pkgs/by-name/ty/typst/with-packages.nix

```
@@ -0,0 +1,48 @@
+{
+  lib,
+  buildEnv,
+  typstPackages,
+  makeBinaryWrapper,
+}:
+
+let
+  typstPackagesFor =
+    typst:
+    lib.makeOverridable (
+      { ... }@typstPkgs:
+      f:
+      let
+        preparedTypstPkgs = f typstPkgs;
+        expandTypstPkgs =
+          ps:
+          if ps == [ ] then
+            ps
+          else
+            lib.foldl' (acc: p: acc ++ lib.singleton p ++ expandTypstPkgs p.typstDeps) [ ] ps;
+      in
+      buildEnv {
+        name = "${typst.name}-env";
+
+        paths = lib.unique (expandTypstPkgs preparedTypstPkgs);
+
+        buildInputs = [ makeBinaryWrapper ];
```

```suggestion
        nativeBuildInputs = [ makeBinaryWrapper ];
```
Again, if it's being executed, it goes to `nativeBuildInputs`. Though I guess in this case it's just a bash function so it doesn't matter too much.